kml/edit-kml-ground-overlay/src/main/java/com/esri/samples/edit_kml_ground_overlay/EditKMLGroundOverlaySample.java

```diff
@@ -0,0 +1,151 @@
+
+  private SceneView sceneView;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene fxScene = new Scene(stackPane);
+      fxScene.getStylesheets().add(getClass().getResource("/style.css").toExternalForm());
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Edit KML Ground Overlay Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(fxScene);
+      stage.show();
+
+      // create a map and add it to the map view
+      ArcGISScene scene = new ArcGISScene(Basemap.createImagery());
+      sceneView = new SceneView();
+      sceneView.setArcGISScene(scene);
+
+      // create a slider for adjusting the rotation
+      Slider slider = new Slider(0, 1, 1);
+      slider.setMaxWidth(300);
+      slider.setShowTickLabels(true);
+      slider.setShowTickMarks(true);
+      // create a label for the slider
+      Label label = new Label("Overlay Opacity: ");
+
+      // create a controls box and add the slider and label
+      HBox controlsHBox = new HBox();
+      controlsHBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.5)"), CornerRadii.EMPTY,
+              Insets.EMPTY)));
+      controlsHBox.setPadding(new Insets(10.0));
+      controlsHBox.setMaxSize(250, 20);
+      controlsHBox.getStyleClass().add("panel-region");
+      controlsHBox.getChildren().addAll(label, slider);
+
+      // create a geometry for the ground overlay
+      Envelope overlayGeometry = new Envelope(-123.066227926904, 44.04736963555683, -123.0796942287304, 44.03878298600624, SpatialReferences.getWgs84());
+
+      // create a KML icon for the overlay image
+      String overlayImageURI = new File("./samples-data/images/1944.jpg").getAbsolutePath();
+      KmlIcon overlayImage = new KmlIcon(overlayImageURI);
+
+      // create the KML ground overlay
+      KmlGroundOverlay kmlGroundOverlay = new KmlGroundOverlay(overlayGeometry, overlayImage);
+
+      // set the rotation of the ground overlay
+      kmlGroundOverlay.setRotation(-3.046024799346924);
+
+      // create a KML dataset with the ground overlay as the root node
+      KmlDataset kmlDataset = new KmlDataset(kmlGroundOverlay);
+
+      // create a KML layer for the scene view
+      KmlLayer kmlLayer = new KmlLayer(kmlDataset);
+
+      // add the KML layer to the scene
+      sceneView.getArcGISScene().getOperationalLayers().add(kmlLayer);
+
+      // move the viewpoint to the ground overlay
+      sceneView.setViewpoint(new Viewpoint(kmlGroundOverlay.getExtent(), new Camera(kmlGroundOverlay.getGeometry().getExtent().getCenter(), 1250, 45, 60, 0)));
+
+      // add an event handler to the slider to apply opacity to the KML ground overlay based on the slider value
+      slider.setOnMouseReleased(event -> kmlGroundOverlay.setColor(ColorUtil.colorToArgb(new Color(0, 0, 0, slider.getValue()))));
```

Comment:
to avoid repetition, you could set up a new Color and apply that to both, e.g.

`Color opacity = new Color(0, 0, 0, slider.getValue());`

`slider.setOnMouseReleased(event -> kmlGroundOverlay.setColor(ColorUtil.colorToArgb(opacity)))`